pandas/core/arrays/datetimelike.py

```diff
@@ -1554,6 +1554,28 @@ def mean(self, skipna=True):
         # Don't have to worry about NA `result`, since no NA went in.
         return self._box_func(result)
 
+    def median(self, axis: Optional[int] = None, skipna: bool = True, *args, **kwargs):
+        nv.validate_median(args, kwargs)
+
+        if axis is not None and abs(axis) >= self.ndim:
+            raise ValueError("abs(axis) must be less than ndim")
+
+        if self.size == 0:
+            if self.ndim == 1 or axis is None:
+                return NaT
```

Comment:
this looks extra complicated 

what is the rationale for handling this way?

Reply:
some of this will end up in nanops.nanmedian mentioned in the OP, but ATM nanmedian casts datetimelike to float (ie copy, not view) so keeping this here for the moment